libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -2813,6 +2814,35 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_curvilinear_compute_smoothness(int meshKernelId, int direction, double* smoothness)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id, {}, does not exist.", meshKernelId);
+            }
+
+            if (meshKernelState[meshKernelId].m_curvilinearGrid == nullptr)
+            {
+                throw meshkernel::MeshKernelError("The curvilinear grid id, {}, does not exist.", meshKernelId);
+            }
+
+            const meshkernel::CurvilinearGrid& grid = *meshKernelState[meshKernelId].m_curvilinearGrid;
+            lin_alg::Matrix<double> smoothnessMatrix;
+
+            meshkernel::CurvilinearGridSmoothness::Compute(grid, direction, smoothnessMatrix);
```

Comment:
you'll need to cast direction from int to SmoothingDirection enum (or whatever you end up calling it). Since we will only allow 0 and 1 and since you can cast a potato if you have the will, throwing an error in `CurvilinearGridSmoothness::Compute` is a good idea. You'll prob need a a ToString enum-to-string converter for use in the exception message.